libraries/core/src/descriptor/mod.rs

```diff
@@ -434,6 +429,14 @@ pub fn resolve_path(source: &str, working_dir: &Path) -> Result<PathBuf> {
         bail!("Could not find source path {}", path.display())
     }
 }
+pub fn source_to_path(source: &str) -> PathBuf {
+    let path = Path::new(&source);
+    if path.extension().is_none() {
+        path.with_extension(EXE_EXTENSION)
+    } else {
+        path.to_owned()
+    }
+}
 
```

Comment:
This looks good. Could be good to have this with working_dir as an optional argument and make this return a result and fail if we have a relative path but we didn't specify a woking dir because we're in remote.

I think we do a lot of check like this

Reply:
Only the number of machines > 1, we use this function, as we only use absolute path, the working_dir doesn't make sense, after the source -> PathBuf in this function, we will check whether the path is relative and give check result. 

https://github.com/dora-rs/dora/blob/9dc07b1e4997948d88dfc47ce3c249904483c4e7/libraries/core/src/descriptor/validate.rs#L37-L44